pkg/kubelet/metrics/collectors/resource_metrics.go

```diff
@@ -0,0 +1,153 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package collectors
```

Comment:
This appears to be very similar to the existing collector.  Can we re-use that one?

Reply:
Yes we can, but that is very limited.

We can only share the `DescribeWithStability`, that is the limited thing i mean, but if we do that, we need a little bit more complex logic  in `CollectWithStability`.  So I think we may lose more than we gain.

Considering the existing will be removed later, we only need to remove dead code by then, and don't need to change any working code.